lib/messages.py

```diff
@@ -196,6 +196,57 @@ def str_func(s):
     namespace = { '__str__': str_func }
   )
 
+#=========================================================================
+# Address message for non-blocking execution mode
+#=========================================================================
+
+def nb_addr(addr_nbits=8, kernel_id_nbits=2, ld_id_nbits=5,
```

Comment:
Can we rename this to `mk_multi_kernel_addr`, imo, this format can be orthogonal to blocking or non-blocking, i.e., even for blocking, we can use this but not check kernel_id nor ld_id, WDYT?